File: src/main/java/io/junseok/mealmate/auth/controller/AuthController.java
package io.junseok.mealmate.auth.controller;

import io.junseok.mealmate.auth.dto.LoginDto;
import io.junseok.mealmate.auth.dto.TokenDto;
import io.junseok.mealmate.auth.service.LoginAuthService;
import javax.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api")
@RequiredArgsConstructor
public class AuthController {
    private final LoginAuthService authService;

    @PostMapping("/login")
    public ResponseEntity<TokenDto> authorize(@Valid @RequestBody LoginDto loginDto) {
        TokenDto tokenDto = authService.login(loginDto);
        return ResponseEntity.ok(tokenDto);
    }
}

File: src/main/java/io/junseok/mealmate/auth/presentation/AuthController.kt
package io.junseok.mealmate.auth.presentation

import io.junseok.mealmate.auth.dto.LoginDto
import io.junseok.mealmate.auth.service.LoginAuthService
import org.jetbrains.annotations.NotNull
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import javax.validation.Valid

@RestController
@RequestMapping("/api")
@CrossOrigin
class AuthController(
    private val loginAuthService: LoginAuthService
) {
    @PostMapping("/login")
    fun authorize(@RequestBody loginDto: LoginDto) =
        loginAuthService.login(loginDto).also {
            ResponseEntity.ok(it)
        }
}